python/_tskitmodule.c

```diff
@@ -9324,31 +9324,53 @@ TreeSequence_get_genotype_matrix(TreeSequence *self, PyObject *args, PyObject *k
         goto out;
     }
     V = (int32_t *) PyArray_DATA(genotype_matrix);
-    vg = PyMem_Malloc(sizeof(tsk_vargen_t));
-    if (vg == NULL) {
+    variant = PyMem_Malloc(sizeof(tsk_variant_t));
```

Comment:
Is there a reason for mallocing the tree and variant rather than using stack storage? It saves a little bit of messing around, and once you make sure to memset them to zero before doing anything it's safe.

Reply:
I've moved it to the stack, couldn't see a reason not. The vargen was on the heap before this diff though.